realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyMediatorGenerator.kt

```diff
@@ -423,7 +423,11 @@ class RealmProxyMediatorGenerator(private val processingEnvironment: ProcessingE
                     Arrays.asList("JSONException")
             )
                 emitMediatorShortCircuitSwitch(writer, emitStatement = { i: Int ->
-                    emitStatement("return clazz.cast(%s.createOrUpdateUsingJsonObject(realm, json, update))", qualifiedProxyClasses[i])
+                    if (!embeddedClass[i]) {
+                        emitStatement("return clazz.cast(%s.createOrUpdateUsingJsonObject(realm, json, update))", qualifiedProxyClasses[i])
+                    } else {
+                        emitStatement("throw new IllegalArgumentException(\"Cannot import orphaned embedded class from json\")")
```

Comment:
Would be `Importing embedded classes from JSON without a parent is now allowed` be more clear? `orphaned` feels like it had a parent once, which it didn't in this case.